src/runner.ts

```diff
@@ -0,0 +1,578 @@
+        for (const name in bowerConfig.devDependencies) {
+          merged.dependencies[name] = bowerConfig.devDependencies[name];
+        }
+      }
+      if (bowerConfig.dependencies) {
+        for (const name in bowerConfig.dependencies) {
+          merged.dependencies[name] = bowerConfig.dependencies[name];
+        }
+      }
+      if (bowerConfig.resolutions) {
+        for (const name in bowerConfig.resolutions) {
+          merged.resolutions[name] = bowerConfig.resolutions[name];
+        }
+      }
+    }
+    return merged;
+  }
+
+  /**
+   * Creates a .bowerrc that tells bower to use the workspace dir (`.`) as
+   * the installation dir (instead of default (`./bower_components`) dir.
+   * Creates a bower.json which sets all the workspace repos as dependencies
+   * and
+   * also includes the devDependencies from all workspace repos under test.
+   */
+  _installWorkspaceDependencies() {
+    const pb =
+        util.standardProgressBar('Installing dependencies with bower...', 1);
+
+    fs.writeFileSync(
+        path.join(this._workspace.dir, '.bowerrc'),
+        JSON.stringify({directory: '.'}));
+
+    const bowerConfig = this._mergedTestRepoBowerConfig();
+
+    // TODO(usergenic): Verify this is even needed.
+    if (!bowerConfig.dependencies['web-component-tester']) {
+      bowerConfig.dependencies['web-component-tester'] = '';
+    }
+
+    // Make bower config point bower packages of workspace repos to themselves
+    // to override whatever any direct or transitive dependencies say.
+    for (const repo of Array.from(this._workspace.repos.entries())) {
+      bowerConfig.dependencies[repo[0]] = `./${repo[1].dir}`;
+    }
+
+    fs.writeFileSync(
+        path.join(this._workspace.dir, 'bower.json'),
+        JSON.stringify(bowerConfig));
+
+    // TODO(usergenic): Can we switch to using bower as library here?  Might
+    // even give us better option for progress bar.
+    // HACK(usergenic): Need a reliable way to obtain the bower bin script.
+    const bowerCmd = path.join(resolve.sync('bower'), '../bin/bower.js');
+    child_process.execSync(`node ${bowerCmd} install -F`, {
+      // node ${bowerCmd} install`, {
+      cwd: this._workspace.dir,
+      stdio: (this._verbose ? 'inherit' : 'ignore')
+    });
+    pb.tick();
+  }
+
+  /**
+   * All repos specified by tests option will be run through wct.
+   */
+  async _testAllTheThings(): Promise<TestResult[]> {
+    if (this._verbose) {
+      console.log('Test all the things...');
+    }
+
+    const testPromises: Promise<TestResult>[] = [];
+
+    for (const repo of Array.from(this._workspace.repos.values())
+             .filter(repo => repo.test)) {
+      try {
+        const testPromise = this._testRateLimiter.schedule(() => {
+          return test(this._workspace, repo, this._wctFlags.split(' '));
+        });
+        testPromises.push(testPromise);
+      } catch (err) {
+        throw new Error(`Error testing ${repo.dir}:\n${err.stack || err}`);
```

Comment:
if this was a mapping with an async mapper, then the errors would be collected up and throw together in promiseAllWithProgress